src/node_buffer.cc

```diff
@@ -19,7 +19,11 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Handle<Value> _obj = argT;                                                \
+  if (!HasInstance(_obj)) {                                                 \
+    return env->ThrowTypeError("must be called on Buffer instance");        \
```

Comment:
OK, so what should I do with this PR? It is no problem for me to rewrite this in @bnoordhuis' style, but if that would not be useful due to the V8 4.4 breakage I can also not bother.

If there is going to be another patch release before V8 4.4 & the rewrite I recommend at least taking my first two commits (the `.toString()` fix and the `.copy()` fix). They are small, don't touch the `ARGS_THIS` macro, and fix problems that are easy to trigger. The copy bug is especially bad because `anyBuffer.copy(obj.misspelledProperty)` segfaults.